### PR TITLE
[ramda] split tests to separate modules: 'pipe' - 'reject'

### PR DESCRIPTION
**Pull request by devrelm**
#37710 attempted to split the tests for `@types/ramda` into separate files per function, but the PR was too large to be approved on its own. This PR is part of a new piecemeal approach to migrating the tests.

I recommend stepping through the PR commit-by-commit to make it easier to check that no individual tests were dropped or changed substantively.

This PR is the next of many to move the functions over in a more piecemeal fashion. It continues from the following PRs:

* #38157 `[ramda] split tests to separate modules - functions beginning with 'a'`
* #38251 `[ramda] split tests to separate modules - functions beginning with 'b' - 'c'`
* #38274 `[ramda] split tests to separate modules - functions beginning with 'd' - 'e'`
* #38347 `[ramda] split tests to separate modules: 'filter' - 'head'`
* #38546 `[ramda] split tests to separate modules: 'identical' - 'isNaN'`
* #38570 `[ramda] split tests to separate modules: 'join' - 'mean'`
* #38615 `[ramda] split tests to separate modules: 'median' - 'none'`
* #40543 `[ramda] split tests to separate modules: 'not' - 'pickBy'`

I'm splitting out the tests into separate files per function to make it easier to track what is and is not being tested. Please read the descriptions for #37710 and #38157 for more details.

*Special Note:*

This particular PR is a duplicate of #40604, which was closed as stale ~6 months ago. It had 2 approvals, so hopefully this one can get approved again and merged in quickly so that I can get the rest of these PRs to get all the remaining tests migrated.

---

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [ ] ~~Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>~~
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~~

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. sequence
1. xor
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fbdb49e2e94f5bbd9d15e4d2ca1822995fa045d0
</p>

File: types/ramda/ramda-tests.ts [hunk 1]
```diff
@@ -474,7 +474,6 @@ type Pair = KeyValuePair<string, number>;
 };
 
 () => {
-    const x          = R.prop("x");
```
Comment (drewwyatt) [inline types/ramda/ramda-tests.ts:701]: What was this doing? anything?

**Review by drewwyatt — APPROVED** [review]
LGTM

File: types/ramda/ramda-tests.ts (hunk 1 above)
Comment (devrelm) [reply types/ramda/ramda-tests.ts:701]: Looks like nothing. Probably initially used in the 4 lines that come after it, but it doesn't seem to do or assert anything here, other than that `R.prop()` *can* take a string and doesn't return `void` or `never`.

**Pull request merged**